lib/SecretsCrypto/sf-secrets-manager.c

```diff
@@ -153,27 +182,66 @@ static gboolean _sf_secrets_manager_perform_interaction(SfSecretsManager *manage
     SfSecretsManagerPrivate *priv = sf_secrets_manager_get_instance_private(manager);
     gchar *id = g_uuid_string_random();
     gboolean *result;
-    SfSecretsInteractionRequest *request = g_object_new(SF_TYPE_SECRETS_INTERACTION_REQUEST,
+    SfSecretsInteractionRequest *request;
+    const gchar *secret_name;
+    const gchar *collection_name;
+    const gchar *plugin_name;
+    const gchar *application_id;
+    gint32 operation;
+    const gchar *authentication_plugin_name;
+    GVariant *prompt_text;
+    GHashTable *pt;
+    gint32 input_type;
+    gint32 echo_mode;
+
+    g_variant_get(args,
+            "(&s&s&s&s(i)&s@a{is}(i)(i))",
+            &secret_name,
+            &collection_name,
+            &plugin_name,
+            &application_id,
+            &operation,
+            &authentication_plugin_name,
+            &prompt_text,
+            &input_type,
+            &echo_mode);
+
+    pt = _sf_prompt_text_from_variant_or_null(prompt_text);
+    g_variant_unref(prompt_text);
+
+    request = g_object_new(SF_TYPE_SECRETS_INTERACTION_REQUEST,
             "id", id,
             "invocation", invocation,
+            "secret-name", secret_name,
+            "collection-name", collection_name,
+            "plugin-name", plugin_name,
+            "application-id", application_id,
+            "operation", operation,
+            "authentication-plugin-name", authentication_plugin_name,
+            "prompt-text", pt,
+            "input-type", input_type,
+            "echo-mode", echo_mode,
             NULL);
-
-    (void)args;
-
     g_object_add_weak_pointer(G_OBJECT(request), (gpointer *)&request);
     g_signal_emit(manager,
             _sf_secrets_manager_signals[SIGNAL_NEW_INTERACTION_REQUEST], 0,
             request,
             &result);
     g_object_unref(request);
 
-    if ((request = g_object_ref(request))) {
+    if (request) {
+        request = g_object_ref(request);
```

Comment:
This explicitly let go of reference and then continue using it, even reacquiring a ref looks really odd (even if the weak ref stuff makes it work ok?). Why not just move the unref below if/else block and avoid murky waters?

Reply:
The point here is to not run the block below, unless someone handling the "new interaction request" takes a ref on the object. It indeed looks kinda shady.